src/unicode.c

```diff
@@ -959,7 +959,7 @@ int cp_name2id(char *encoding)
 	char *d = enc;
 
 	if (!encoding || strlen(encoding) > sizeof(enc))
-		return CP_UNDEF;
+		goto err;
```

Comment:
This might lead to regression problems. If encoding == NULL we should still return CP_UNDEF.